cmd/publishing-bot/publisher.go

```diff
@@ -71,472 +42,45 @@ type PublisherMunger struct {
-				dst: coordinate{repo: "metrics", branch: "master", dir: "./"},
-				deps: []coordinate{
-					{repo: "apimachinery", branch: "master"},
-					{repo: "api", branch: "master"},
-					{repo: "client-go", branch: "master"},
-				},
-			},
-			{
-				// rule for the sample-apiserver 1.7 branch
-				src: coordinate{repo: config.SourceRepo, branch: "release-1.7", dir: "staging/src/k8s.io/metrics"},
-				dst: coordinate{repo: "metrics", branch: "release-1.7", dir: "./"},
-				deps: []coordinate{
-					{repo: "apimachinery", branch: "release-1.7"},
-					{repo: "client-go", branch: "release-4.0"},
-				},
-			},
-			{
-				// rule for the sample-apiserver 1.8 branch
-				src: coordinate{repo: config.SourceRepo, branch: "release-1.8", dir: "staging/src/k8s.io/metrics"},
-				dst: coordinate{repo: "metrics", branch: "release-1.8", dir: "./"},
-				deps: []coordinate{
-					{repo: "apimachinery", branch: "release-1.8"},
-					{repo: "api", branch: "release-1.8"},
-					{repo: "client-go", branch: "release-5.0"},
-				},
-			},
-			{
-				// rule for the sample-apiserver 1.9 branch
-				src: coordinate{repo: config.SourceRepo, branch: "release-1.9", dir: "staging/src/k8s.io/metrics"},
-				dst: coordinate{repo: "metrics", branch: "release-1.9", dir: "./"},
-				deps: []coordinate{
-					{repo: "apimachinery", branch: "release-1.9"},
-					{repo: "api", branch: "release-1.9"},
-					{repo: "client-go", branch: "release-6.0"},
-				},
-			},
-		},
-		publishScript: "/publish_scripts/publish_metrics.sh",
+func (p *PublisherMunger) readRepositoryRules(cfg *config.Config) error {
+	// If RULE_FILE_PATH is detected, check if the source repository include rules files.
+	if len(os.Getenv("RULE_FILE_PATH")) > 0 {
+		cfg.RulesFile = filepath.Join(p.baseRepoPath, cfg.SourceRepo, os.Getenv("RULE_FILE_PATH"))
 	}
 
-	codeGenerator := repoRules{
-		dstRepo: "code-generator",
-		srcToDst: []branchRule{
-			{
-				// rule for the metrics master branch
-				src:  coordinate{repo: config.SourceRepo, branch: "master", dir: "staging/src/k8s.io/code-generator"},
-				dst:  coordinate{repo: "code-generator", branch: "master", dir: "./"},
-				deps: []coordinate{},
-			},
-			{
-				// rule for the metrics release-1.8 branch
-				src:  coordinate{repo: config.SourceRepo, branch: "release-1.8", dir: "staging/src/k8s.io/code-generator"},
-				dst:  coordinate{repo: "code-generator", branch: "release-1.8", dir: "./"},
-				deps: []coordinate{},
-			},
-			{
-				// rule for the metrics release-1.9 branch
-				src:  coordinate{repo: config.SourceRepo, branch: "release-1.9", dir: "staging/src/k8s.io/code-generator"},
-				dst:  coordinate{repo: "code-generator", branch: "release-1.9", dir: "./"},
-				deps: []coordinate{},
-			},
-		},
-		publishScript: "/publish_scripts/publish_code_generator.sh",
+	// NOTE: Order of the repos is sensitive!!!
+	// A dependent repo needs to be published first, so that other repos can vendor its latest revision.
+	rules, err := config.LoadRules(cfg)
```

Comment:
just do this call directly for New. Seems to be overkill to have a func that call essentially one func.